src/CTA.WebForms2Blazor/Services/WorkspaceManagerService.cs

```diff
@@ -0,0 +1,153 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Text;
+
+namespace CTA.WebForms2Blazor.Services
+{
+    public class WorkspaceManagerService
+    {
+        private const string WORKSPACE_DUPLICATE_ERROR = "Attempted to create a new Blazor workspace when one already exists";
+        private const string WORKSPACE_MISSING_ERROR_TEMPLATE = "Workspace {0} operation attempted, but no workspace exists";
+        private const string WORKSPACE_UPDATE_FAILED_ERROR_TEMPLATE = "Workspace {0} operation attempted, but workspace failed to apply changes";
+
+        // We track number of projects and documents explicitly because the number of
+        // documents in the workspace only equates to code files, and also for use by
+        // functions or services that wait on conditions using these values. Continually
+        // enumerating the project list and the individual document lists would add up
+        // computationally during these waiting periods
+        private int _numProjects;
+        private int _numDocuments;
+        private int _expectedProjects;
+        private int _expectedDocuments;
+        private AdhocWorkspace _workspace;
+
+        public Solution CurrentSolution { get { return _workspace.CurrentSolution; } }
+
+        public void CreateSolutionFile()
+        {
+            if (_workspace != null)
+            {
+                throw new InvalidOperationException(WORKSPACE_DUPLICATE_ERROR);
+            }
+
+            _workspace = new AdhocWorkspace();
```

Comment:
What is an AdhocWorkspace?

Reply:
It's basically a container for solutions, projects, and code documents that we can use to create syntax trees and semantic models. The other workspace type we could use is the MSBuildWorkspace which would be better for the WebForms project but its not available in netcore3.1 which cta uses. Adhoc is nice because it allows full modification of solutions and projecs though, which MSBuild doesn't. Here are the MS docs on Adhoc workspaces https://docs.microsoft.com/en-us/dotnet/api/microsoft.codeanalysis.adhocworkspace?view=roslyn-dotnet-3.9.0